dashboard/modules/job/tests/test_job.py

```diff
@@ -1,23 +1,165 @@
+        actors.append(Actor.remote(x))
+
+    counter = 0
+    while True:
+        for a in actors:
+            r = a.foo.remote(counter)
+            print(ray.get(r))
+            counter += 1
+            time.sleep(1)
+
+
+if __name__ == "__main__":
+    ray.init()
+    main()
+"""
+
+
+def _gen_job_zip(job_code, driver_entry):
+    with tempfile.NamedTemporaryFile(suffix=".zip", delete=False) as f:
+        with zipfile.ZipFile(f, mode="w") as zip_f:
+            with zip_f.open(f"{driver_entry}.py", "w") as driver:
+                driver.write(job_code.encode())
+        return f.name
+
+
+def _gen_url(web_url, path):
+    return f"{web_url}/test/file?path={path}"
+
+
+def _get_python_job(web_url):
+    driver_entry = "simple_job"
+    path = _gen_job_zip(TEST_PYTHON_JOB_CODE, driver_entry)
+    url = _gen_url(web_url, path)
+    job = copy.deepcopy(TEST_PYTHON_JOB)
+    job["url"] = url
+    job["driverEntry"] = driver_entry
+    return job
+
+
+@pytest.mark.parametrize(
+    "ray_start_with_dashboard", [{
+        "job_config": ray.job_config.JobConfig(code_search_path=[""]),
+    }],
+    indirect=True)
+def test_submit_job(disable_aiohttp_cache, enable_test_module,
+                    ray_start_with_dashboard):
+    assert (wait_until_server_available(ray_start_with_dashboard["webui_url"])
+            is True)
+    webui_url = ray_start_with_dashboard["webui_url"]
+    webui_url = format_web_url(webui_url)
+
+    shutil.rmtree(JOB_ROOT_DIR, ignore_errors=True)
+
+    job_id = None
+
+    def _check_running():
+        nonlocal job_id
+        if job_id is None:
+            resp = requests.post(
+                f"{webui_url}/jobs", json=_get_python_job(webui_url))
+            resp.raise_for_status()
+            result = resp.json()
+            assert result["result"] is True, resp.text
+            job_id = result["data"]["jobId"]
+
+        resp = requests.get(f"{webui_url}/jobs?view=summary")
+        resp.raise_for_status()
+        result = resp.json()
+        assert result["result"] is True, resp.text
+        summary = result["data"]["summary"]
+        assert len(summary) == 2
+
+        resp = requests.get(f"{webui_url}/jobs/{job_id}")
+        resp.raise_for_status()
+        result = resp.json()
+        assert result["result"] is True, resp.text
+        job_info = result["data"]["detail"]["jobInfo"]
+        assert job_info["jobId"] == job_id
+
+        resp = requests.get(f"{webui_url}/jobs/{job_id}")
+        resp.raise_for_status()
+        result = resp.json()
+        assert result["result"] is True, resp.text
+        job_info = result["data"]["detail"]["jobInfo"]
+        assert job_info["isDead"] is False
+        job_actors = result["data"]["detail"]["jobActors"]
+        job_workers = result["data"]["detail"]["jobWorkers"]
+        assert len(job_actors) > 0
+        assert len(job_workers) > 0
+
+    timeout_seconds = 30
+    start_time = time.time()
+    last_ex = None
+    while True:
```

Comment:
use `wait_for_condition`?

Reply:
I updated the `wait_until_succeeded_without_exception` to throw the last exception when timeout.